src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs

```diff
@@ -188,6 +209,10 @@ public class Frame<TContext> : Frame
                 // SetBadRequestState logs the error.
                 SetBadRequestState(ex);
             }
+            catch (IOException ex) when (ex.InnerException is UvException)
```

Comment:
Why don't you do all checks here and avoid the new `try`/`catch` block around the code that reads the request line?